### PR TITLE
Add link to forked repo below the source

### LINKED ISSUE #1129: Add link to forked repo below the original
To resolve this issue, jump to: https://github.com/sindresorhus/refined-github/issues/1129#issuecomment-371044209

[Issuehunt badges](https://issuehunt.io/r/sindresorhus/refined-github/issues/1129)

Hey guys! Amazing addon with lot of awesome features 💯 

Github displays the original repo path when visiting a fork. But, the original don't have link to your fork. Right now we have to access the forks from here... 

My proposal is to add a `forked to cg-cnu/refined-github` below the original repo.

Let me know your comments and ideas. Will be happy to do a pr if you guys like it. 
Cheers 😄

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by jerone**
### Description
Add links to the forked repos below the original repo. For users and organizations.

### Closes
Closes #1129
Supersedes attempt 1 #1108
Supersedes attempt 2 #2105
Supersedes attempt 3 #2141

### Screenshots
**Original repo:**
forked-to

**User forked repo:**
image

**Organization forked repo:**
image

### Test
1. Open any repo and click on the <kbd>Fork</kbd> button.
2. Fork the repo.
3. Go back to the original repo. 
4. Below the repo name, a link to your fork will presented.

### CODE REVIEW

**Comment by jerone** [comment]
@bfred-it Clean up even more; hope it satisfies. Feel free to edit at your will.

File: source/libs/page-detect.ts [hunk 1]
```diff
@@ -89,6 +89,8 @@ export const isRepoSettings = (): boolean => /^settings/.test(getRepoPath()!);
 
 export const isRepoTree = (): boolean => isRepoRoot() || /^tree\//.test(getRepoPath()!);
 
+export const isOwnRepo = (): boolean => isRepo() && select.exists('.reponav-item[href$="/settings"]');
```
Comment (sindresorhus) [inline source/libs/page-detect.ts]: `select.exists('.reponav-item[href$="/settings"]')` is true even for organization repos that are technically not "own". Depending on the use-case here, it should either be renamed to `isRepoWithAccess` (better name please) or made more strict.

File: source/features/forked-to.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+	return undefined;
+}
+
+function getSourceRepo(): string {
+	return findForkedRepo() || getRepoURL();
+}
+
+async function validateFork(repo: string): Promise<boolean> {
+	const response = await fetch(location.origin + '/' + repo, {method: 'HEAD'});
```
Comment (sindresorhus) [inline source/features/forked-to.tsx]:
```suggestion
	const response = await fetch(`${location.origin}/${repo}`, {method: 'HEAD'});
```

File: source/features/forked-to.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			return forkElement.parentNode!.textContent!.trim();
+		});
+		addAndStoreCache(getSourceRepo(), ...forks);
+	});
+}
+
+function findForkedRepo(): string | undefined {
+	const forkSourceElement = select<HTMLAnchorElement>('.fork-flag:not(.rgh-forked) a');
+	if (forkSourceElement) {
+		return forkSourceElement.pathname.substring(1);
```
Comment (sindresorhus) [inline source/features/forked-to.tsx]:
```suggestion
		return forkSourceElement.pathname.slice(1);
```

Just my preference. https://github.com/sindresorhus/eslint-plugin-unicorn/issues/65

File: source/libs/page-detect.ts (hunk 1 above)
Comment (jerone) [reply source/libs/page-detect.ts]: It should detect orgs also, so `isRepoWithAccess` is a good alternative name.

**Comment by bfred-it** [comment]
Not a good idea to include the full issue number in the branch name, [this happens](https://github.com/sindresorhus/refined-github/issues/1129#ref-commit-9df5008): 😅

Screenshot 2019-06-23 at 15 41 55

**Comment by jerone** [comment]
Will try to remember

**Comment by sindresorhus** [comment]
The implementation looks good to me. Can you mention the feature in the readme with a link to a screenshot?

File: source/features/forked-to.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	watchForkDialog();
+
+	rememberCurrentFork();
+
+	showForks();
+}
+
+features.add({
+	id: 'forked-to',
+	description: 'Add link to forked repo below the source',
```
Comment (sindresorhus) [inline source/features/forked-to.tsx]: Can you add the `screenshot` property?

**Comment by jerone** [comment]
[**@sindresorhus**](https://github.com/sindresorhus) commented on [Jul 1, 2019, 8:46 AM GMT+2](https://github.com/sindresorhus/refined-github/pull/2153#issuecomment-507137413 "2019-07-01T06:46:05Z - Replied by Github Reply Comments"):
> The implementation looks good to me. Can you mention the feature in the readme with a link to a screenshot?

Done.

File: readme.md [hunk 5]
```diff
@@ -143,6 +143,7 @@ GitHub Enterprise is also supported. More info in the options.
 - [SVG files in a PR default to rich-diff view.](https://user-images.githubusercontent.com/5243867/57125552-c08a2b00-6d81-11e9-9b84-cdb535baa98e.png)
 - [Download count next to release assets.](https://user-images.githubusercontent.com/14323370/58944460-e1aeb480-874f-11e9-8052-2d4dc794ecab.png)
 - [The most useful comment in issues is highlighted.](https://user-images.githubusercontent.com/1402241/58757449-5b238880-853f-11e9-9526-e86c41a32f00.png)
+- [Add link to forked repo below the source.](https://user-images.githubusercontent.com/55841/60462644-25f77c80-9c4a-11e9-8c69-582d00dd75dd.png)
```
Comment (bfred-it) [inline readme.md]: * Your repo forks are shown under the repo title. 

The screenshot should be retina

Comment (jerone) [reply readme.md]: Done.

**Comment by sindresorhus** [comment]
Thanks for contributing this feature 🙌

**Comment by bfred-it** [comment]
<3

**Comment by 1138-4EB** [comment]
@jerone, first off, thanks for contributing this feature! ❤️ 

I'm using [Firefox Multi-Account Containers](https://addons.mozilla.org/es/firefox/addon/multi-account-containers/), to handle two different GitHub accounts. However, this new feature ignores 'container isolation'. I.e., if the same repo is cloned in both accounts, I see both of them listed when I visit the original repo. I would expect a single fork to be shown instead, the one that corresponds to the account that is logged in. More precisely:

- No container: logged in with `user_a`, which does NOT belong to `org`.
- Container `Personal`: logged in with `user_b`, which does belong to `org`.

When visiting `orig/repo`, I see:

- forked to user_a/repo
- forked to org/repo

Instead, I would expect to see `forked to user_a/repo` only, when browsing in 'No container', and `forked to org/repo` only when browsing in 'Personal'. This is because following the link to the fork does not change the container. The user needs to explicitly right-click to open it in the corresponding container.

Is this expected by design?

**Comment by jerone** [comment]
Interesting issue. You are encountering this issue, because all forks are saved in a cache storage. Cache is used on more features, but those are not account depended, like this feature. The solution would be to use a cache key which also contains the logged-in username.

@bfred-it @sindresorhus Is multiple accounts something we want to support?

**Comment by bfred-it** [comment]
@jerone this can probably be solved by including the current username in the cache key itself, e.g. `forked-to:<currentuser>:<user>/<repo>`

**Comment by jerone** [comment]
@bfred-it Yes, exactly like I said 😄 

> The solution would be to use a cache key which also contains the logged-in username.

**Comment by 1138-4EB** [comment]
@bfred-it, @jerone, do you mean in https://github.com/sindresorhus/refined-github/blob/master/source/features/forked-to.tsx#L8? Would it be `forked-to:${currentuser}:${user}/${repo}`?

**Comment by jerone** [comment]
@1138-4EB Yeah. I'm testing right now. Will send PR soon.

**Comment by jerone** [comment]
PR here: https://github.com/sindresorhus/refined-github/pull/2215

**Pull request merged**